apps/webapp/src/state/swap.ts

```diff
@@ -90,26 +96,20 @@ export const createSwapSlice = (): SliceCreator<SwapSlice> => (set, get) => {
 };
 
 const assembleSwapRequest = async ({ assetIn, amount, assetOut }: SwapSlice) => {
-  if (assetIn?.value.valueView.case !== 'knownAssetId') throw new Error('unknown denom selected');
-  if (!assetIn.value.valueView.value.metadata?.penumbraAssetId)
-    throw new Error('missing metadata for assetIn');
-  if (assetIn.address.addressView.case !== 'decoded')
-    throw new Error('address in view is not decoded');
-  if (!assetIn.address.addressView.value.index) throw new Error('No index for assetIn address');
-  if (assetOut?.penumbraAssetId === undefined) throw new Error('assetOut has no asset id');
+  if (!assetIn) throw new Error('`assetIn` was undefined');
 
   return new TransactionPlannerRequest({
     swaps: [
       {
-        targetAsset: assetOut.penumbraAssetId,
+        targetAsset: getAssetId.orThrow(assetOut),
         value: {
           amount: toBaseUnit(
             BigNumber(amount),
-            getDisplayDenomExponent(assetIn.value.valueView.value.metadata),
+            getMetadata.pipe(getDisplayDenomExponent).orThrow(assetIn.value),
           ),
-          assetId: assetIn.value.valueView.value.metadata.penumbraAssetId,
+          assetId: getMetadata.pipe(getAssetId).orThrow(assetIn.value),
```

Comment:
Reviewers, if you don't like inlining the `pipe`ing here, I can also extract these above, e.g., 

```ts
const getAssetIdFromValueView = getMetadata.pipe(getAssetId).orThrow

return new TransactionPlannerRequest({
  swaps: [
    {
      value: {
        // ...
        assetId: getAssetIdFromValueView(assetIn.value)
      },
    }
})
```